Yaya/custom recycler view+bottom nav bar+favorites page+expanded list+refresh button+Search functions

lamsalab opened the pull request:
I think this is a minimum viable product. I have tested it on multiple devices and it works fine. I still have more to implement, which will be shipped periodically unlike this one. Let me know what I have to change before we make a merge.

lamsalab committed:
Made the basic layout for individual event

lamsalab committed:
Added the page for individual event

lamsalab committed:
Added the bottom navigation bar

lamsalab committed:
Added Recycler View, Made new activity for indivdual event page

lamsalab committed:
Limited character width in title, limited to only portrait mode

lamsalab committed:
Added ability to add events to calenders. Still not check it for multiple devices.

lamsalab committed:
Added email field for contacts. Added person name in contacts as well

lamsalab committed:
Bottom navigation fragmnets seem to work now. But the soln is hacky.

lamsalab committed:
Set up favorites button. Seems to work

lamsalab committed:
Added persistant storage

lamsalab committed:
Favorites are restored

lamsalab committed:
Added padding to recycler view (top and bottom)

lamsalab committed:
Added app logo, fixed issue with toggle getting activated while scrolling

lamsalab committed:
Added favorites layout. Events are retreived even after a refresh.

lamsalab committed:
Refactored some code and removed bottom navigation bar animation for now

lamsalab committed:
Added swipe to delete in favorites. But still a bug. Recycler view doesnt refresh. Changing the way adapters are managed.

lamsalab committed:
Swipe to delete works now. Still need to refactor code in async parser

lamsalab committed:
Implemented the search function. There is a bug with if query doesnt exist

lamsalab committed:
Added unversioned files

lamsalab committed:
Fixed the bug with search going out of bounds.

lamsalab committed:
Customized the recycler view by adding a info row for each day. Further factored code by adding comments and
cutting down lines that are too long. Also, changed the search page to calender (yet to be implemented)

lamsalab committed:
Added snackbar to undo deletion with swipe for favorites items

lamsalab committed:
Added ripple effect when clicked in recycler view. Added documentation for snackbar as well

lamsalab committed:
Fixed bugs with creating a new activity for each event click and restoring item with snackbar, when list size is 1

lamsalab committed:
Added overflow button for setting and help. Need to implement

lamsalab committed:
Added option to send feedback through email

lamsalab committed:
Made a separate page for feedback

lamsalab committed:
Added the setting layout. Did some refactoring which included adding variables to dimens

lamsalab committed:
Added map functionality

lamsalab committed:
Recycler view state is restored

lamsalab committed:
Updated gradle

lamsalab committed:
Changed some variable names

lamsalab committed:
Changed from shared preferences to writing to a file. Also chnaged some
mismatched variables names.

lamsalab committed:
Removed Bottom navigation hiding. Not necessary.

lamsalab committed:
Added back button and changed the toolbar icons to be white to match
the overall design.

lamsalab committed:
Added more robustness to the parser since some events did not specify time.
Fixed the bug with Swipe to refresh. Now the Progress dialog box only
shows up with fresh starts of the app or if you click the refresh button.
Add splash screen just for the app startup time.

lamsalab committed:
Changed scopes of some methods to private

lamsalab committed:
Changed the layout of individual event profile for better aesthetics

lamsalab committed:
Fixed the bug with progress dialog showing up while swiping down to refresh

lamsalab committed:
Added notification feature. Notifies if event from favorites is going on
Need to add ability to delete scheduled notification if the user removes from favorites.

lamsalab committed:
Notification deletion works when event is removed from favorites

lamsalab committed:
Changed the favorites layout sightly

lamsalab committed:
Added more markers in the map
Added toolbar to feedback page. Added back button as well
Made sure notification is set only if the event is after current time

mattorib commented on app/src/main/java/edu/grinnell/appdev/events/FragmentMap.java:
```diff
@@ -0,0 +1,142 @@
+
+        }
+
+        mapView.getMapAsync(new OnMapReadyCallback() {
```
There seems to be an authentication issue which causes the map to not work at all.

lamsalab replied:
I tested it and it seems to work on multiple devices.

mattorib commented on app/src/main/java/edu/grinnell/appdev/events/AsyncParser.java:
```diff
@@ -19,35 +22,58 @@
+    private Long startTime;
+    private Long endTime;
+    private String orgEmail;
+    private String organizer;
```
Unused variable

mattorib commented on app/src/main/java/edu/grinnell/appdev/events/AsyncParser.java:
```diff
@@ -59,30 +85,42 @@ protected Integer doInBackground(String... strings) {
             return ERROR_PARSING;
         }
-
+        int count = 0;
```
Unused variable

mattorib commented on app/src/main/java/edu/grinnell/appdev/events/onParseComplete.java:
```diff
@@ -3,6 +3,7 @@
 import java.util.List;
 
+
 /**
```
Classes and interfaces should be named with upper camelcase, so the first letter should be capitalized.

mattorib reviewed (changes requested):
Looks good for the most part! The map needs an authentication token, which you should be able to produce, and the files should be organized a little more (ie. adapters, activities, models, etc. in individual folders).

lamsalab committed:
Added submitter name, submitter email, and organizer to individual
events page

lamsalab committed:
Refactored the code into various folders.
Made all the necessary changes.
The map has been tested on multiple devices and it works.

lamsalab committed:
Fixed a bug with toolbar for feedback activity

zoerosemarie reviewed (approved):
Looks good!
